github/github.go

```diff
@@ -687,7 +687,7 @@ func CheckResponse(r *http.Response) error {
 			Response: errorResponse.Response,
 			Message:  errorResponse.Message,
 		}
-	case r.StatusCode == http.StatusForbidden && errorResponse.DocumentationURL == "https://developer.github.com/v3/#abuse-rate-limits":
+	case r.StatusCode == http.StatusForbidden && (strings.HasPrefix(errorResponse.Message, "You have triggered an abuse detection mechanism") || strings.HasSuffix(errorResponse.DocumentationURL, "#abuse-rate-limits")):
```

Comment:
I think it'd be better to simplify the detection check to just:

```Go
case r.StatusCode == http.StatusForbidden && strings.HasSuffix(errorResponse.DocumentationURL, "/v3/#abuse-rate-limits"):
```

It would handle both enterprise and not situations. False positives and false negatives are unlikely (unless GitHub changes things on their end, but then we can update our code accordingly). It would be easier to understand and maintain.